### PR TITLE
add car price + purchase validation

### PR DESCRIPTION
**Pull request by MarkLemashov**

### CODE REVIEW

File: cars/app/cars/models.py [hunk 1]
```diff
@@ -7,16 +8,36 @@ class Customer(models.Model):
...
     def buy_car(self, car_id):
         car = Car.objects.get(id=car_id)
-        car.customer = self;
-        car.save()
+        self.money -= car.price
+        try:
+            self.save()
+            car.customer = self;
+            car.save()
+        except Exception as error:
```
Comment (michael-l-a) [inline cars/app/cars/models.py]: Exception is redundant here. 
You use exceptions on where you don't control flow fully. 
For example, when calling database, API of other services and so on. 
If the code is 100% your you know what you transmit as a parameter and what you do with it. Exception will never happen. 

You sae in DB right here. But DB library that you use already has this Exception. You don't need to wrap it up.

File: cars/app/cars/models.py [hunk 2]
```diff
@@ -7,16 +8,36 @@ class Customer(models.Model):
 
     def buy_car(self, car_id):
         car = Car.objects.get(id=car_id)
-        car.customer = self;
-        car.save()
+        self.money -= car.price
```
Comment (michael-l-a) [inline cars/app/cars/models.py]: Use separate if-statement for the balance check. 
Raise if car.price is higher than the current balance value.

if something:
   raise something 
____

One important thing: use it in signal, not here

File: cars/app/cars/models.py [hunk 3]
```diff
@@ -7,16 +8,36 @@ class Customer(models.Model):
...
+        self.save()
 
     def __str__(self):
         return self.name
 
 
 class Car(models.Model):
-    make = models.CharField(max_length=60)
+    make = models.CharField(max_length=70)
+    price = models.IntegerField(default=0)
```
Comment (michael-l-a) [inline cars/app/cars/models.py]: car never costs 0 money. This parameter should be mandatory.

Setting default value means "I am Ok if you don't provide a value for me, I will set something default".

File: cars/app/cars/models.py [hunk 4]
```diff
@@ -7,16 +8,36 @@ class Customer(models.Model):
...
+            self.save()
+            car.customer = self;
+            car.save()
+        except Exception as error:
+            print(error)
+            self.money = Customer.objects.get(id=self.id).money
+            self.save()
+
+    def earn(self, money_earned):
+        self.money += money_earned
```
Comment (michael-l-a) [inline cars/app/cars/models.py:20]: What is the initial value of self.money?

**Review by michael-l-a — CHANGES_REQUESTED** [review]

File: cars/app/cars/models.py [hunk 5]
```diff
@@ -5,12 +5,14 @@
...
 
     def buy_car(self, car_id):
         car = Car.objects.get(id=car_id)
-        self.item_to_purchase = car
-        self.save()
+        if self.money >= car.price:
+            self.money -= car.price
+            self.save()
+            car.customer = self
+            car.save()
```
Comment (michael-l-a) [inline cars/app/cars/models.py:15]:
```
else:
    raise RuntimeError('not enough money on the account')
```

Comment (MarkLemashov) [reply cars/app/cars/models.py:15]: fixed.

**Review by michael-l-a — APPROVED** [review]

**Pull request merged**